### DISCUSSION TITLE
建议为自动展开弹幕列表功能设置最大弹幕数量限制

### DISCUSSION THREAD

**Discussion by PriceHu** [the1812/Bilibili-Evolved#2972] [功能建议 / Ideas]
#### 针对功能

自动展开弹幕列表 / expandDanmakuList

#### 建议内容

为该功能增加一个最大弹幕数量限制，如当前视频弹幕数大于限制时，不自动展开

#### 建议理由

- 当视频的弹幕数量过大时，展开弹幕列表会造成页面严重卡顿。示例视频：[2008年北京奥运会开幕式](https://www.bilibili.com/video/BV1AN411Z7wf)。猜测原因是播放器加载弹幕是分段进行的，因此页面加载时只需请求少量数据，但展开弹幕列表后将请求所有弹幕，当弹幕总数过大时将严重消耗资源。
- 本人不在国内网络环境，需要测试排除网络问题。但根据Chromium任务管理器显示，展开弹幕列表加载弹幕时，当前标签页CPU占用达到100，内存占用不断上升，但网络活动很少。

**Comment by MC-dusk** [comment]
失效了
